lib/blocks.php

```diff
@@ -296,3 +296,219 @@ function gutenberg_enqueue_editor_block_styles_assets() {
 	}
 	add_action( 'enqueue_block_editor_assets', 'gutenberg_enqueue_editor_block_styles_assets' );
 }
+
+/**
+ * Renders the classNames and styles for blocks
+ *
+ * @param string $block_content Output of the current block.
+ * @param array  $block Block Object.
+ * @return string New block output.
+ */
+function gutenberg_experimental_apply_classnames_and_styles( $block_content, $block ) {
+	// Don't filter template part blocks since we filter the blocks in each template
+	// part individually.
+	if ( 'core/template-part' === $block['blockName'] ) {
+		return $block_content;
+	}
+
+	if ( isset( $block['attrs'] ) ) {
+		// Check what style features the block supports.
+		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+		if ( $block_type ) {
+			$supports = gutenberg_experimental_global_styles_get_supported_styles( $block_type->supports );
+		} else {
+			$supports = array();
+		}
+		// Return early if nothing is supported.
+		if ( sizeof( $supports ) === 0 ) {
+			return $block_content;
+		}
+
+		$colors     = gutenberg_experimental_build_css_colors( $block['attrs'], $supports );
+		$typography = gutenberg_experimental_build_css_typography( $block['attrs'], $supports );
+
+		$extra_classes = array_merge(
+			$colors['css_classes'],
+			$typography['css_classes'],
+			isset( $block['attrs']['className'] ) ? array( $block['attrs']['className'] ) : array(),
+			isset( $block['attrs']['align'] ) ? array( 'has-text-align-' . $block['attrs']['align'] ) : array()
+		);
+		$extra_styles  = (
+			$colors['inline_styles'] ||
+			$typography['inline_styles']
+		) ? esc_attr( $colors['inline_styles'] ) .
+			esc_attr( $typography['inline_styles'] )
```

Comment:
should these be done separately? E.g. if `$colors['inline_styles']` is null, we still try to escape it if `$typography['inline_styles']` is truthy

Reply:
At the very least they are empty strings.  I don't think escaping 1 empty string is a large concern.  Separating the steps and checking / concatenating independently may just be redundant? 

 In fact on the opposite end I wonder if just getting rid of the ternary and doing `$extra_styles = esc_attr( $colors ) . esc_attr( $typography );` would make sense?  If they are always empty strings falling back to an empty string seems redundant.